source/adapters/level_zero/event.cpp

```diff
@@ -882,6 +882,12 @@ urEventRelease(ur_event_handle_t Event ///< [in] handle of the event object
 ) {
   Event->RefCountExternal--;
   UR_CALL(urEventReleaseInternal(Event));
+  // If this is a Completed Event Wait Out Event, then we need to cleanup the
+  // event at user release and not at the time of completion.
+  if (Event->CommandType == UR_COMMAND_EVENTS_WAIT && Event->Completed) {
```

Comment:
isn't this a potential use-after-free?

Reply:
I retested including while disabling the event caching and I found that the additional release was not needed, just the cleanup call.

Please re-review, the updated changed still solves the issue, while ensuring no use after free. The fix ensures that the event is "cleaned up" after it has been released by the user in this case instead of having a floating event with no reuse or destroy. internally, the cleanup call calls release internal such that the event is either destroyed or is added to the cache now that the user is done with it.